components/camera/rtsp/rtsp.go

```diff
@@ -0,0 +1,240 @@
+// Package rtsp implements an RTSP camera client for RDK
+package rtsp
+
+import (
+	"context"
+	"image"
+	"io"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/aler9/gortsplib/v2"
+	"github.com/aler9/gortsplib/v2/pkg/base"
+	"github.com/aler9/gortsplib/v2/pkg/liberrors"
+	"github.com/aler9/gortsplib/v2/pkg/url"
+	"github.com/edaniels/golog"
+	"github.com/edaniels/gostream"
+	"github.com/pion/rtp"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/rimage/transform"
+	"go.viam.com/rdk/utils"
+)
+
+var (
+	model            = resource.NewDefaultModel("rtsp")
+	clientTerminated = liberrors.ErrClientTerminated{}
```

Comment:
mostly because on line 122, I do the re-connectivity check every 5 sec, which looks for this error. I felt like there wasn't a need to re-declare that error every 5 seconds for the check.